presto-main/src/main/java/com/facebook/presto/operator/aggregation/AggregationCompiler.java

```diff
@@ -151,7 +152,7 @@ public static BindableAggregationFunction generateBindableAggregationFunction(Cl
                     }
 
                     genericSignature = genericSignature.isPresent() ? genericSignature : Optional.of(implementation.getSignature());
-                    // TODO validateSignature(signature, implementation.getSignature());
+                    validateSignaturesCompatibility(genericSignature, implementation.getSignature());
```

Comment:
We only enforce matching (not same) signatures for generic and specialized implementations. There is a generic signature for all implementations but some implementation may have more specialized variants of it. The important bit is that all implementations must match the generic one.

This is abstracted a little bit more to the ParamtericImplementations class in https://github.com/Teradata/presto/pull/396/commits/4f0603b00f88170fbc7ae338c6ca049cd952d8cb#diff-705206c735c150e2452b07142f838e95R115